### ISSUE TITLE
False Positive "AxiosError is defined but never used" no-unused-vars

### ISSUE THREAD

**Issue by StrangeWill** [vuejs/eslint-plugin-vue#1041]
**Tell us about your environment**	
* **ESLint version: 6.8.0**	
* **eslint-plugin-vue version: 6.1.2**	
* **Node version: 12.14.1**	

**Please show your full configuration:**

```json
module.exports = {
    root: true,
    env: {
        node: true
    },
    'extends': [
        'plugin:vue/essential',
        'eslint:recommended',
        '@vue/typescript'
    ],
    rules: {
        'no-console': process.env.NODE_ENV === 'production' ? 'error' : 'off',
        'no-debugger': process.env.NODE_ENV === 'production' ? 'error' : 'off'
    },
    parserOptions: {
        parser: '@typescript-eslint/parser'
    },
    overrides: [
        {
            files: [
                '**/__tests__/*.{j,t}s?(x)'
            ],
            env: {
                mocha: true
            }
        }
    ]
}

```

**What did you do?**	

```vue
<script lang="ts">
// ...
import { AxiosError } from "axios";
// ...

const response = await this.$http
	.post<JwtToken>(
		"authentication_token",
		{
			email: this.username,
			password: this.password
		},
		{
			headers: {
				authorization: ""
			}
		}
	)
	.catch((error: AxiosError) => { // <-- this line right here
		if (error?.response?.status == 401) {
			throw "Login Failed";
		}

		throw error?.message;
	});
// ...
```

**What did you expect to happen?**

Should pass es-lint fine

**What actually happened?**

It *sometimes* passes es-lint, appears to do it on first pass, but fails on subsequent passes when running `npm run serve`.

**Comment by armano2** [comment]
This is not issue in this plugin, as types are handled by https://github.com/typescript-eslint/typescript-eslint

**Comment by StrangeWill** [comment]
Thanks, I'll take it up over there.
